pipelines/build/common/build_base_file.groovy

```diff
@@ -1014,17 +1020,20 @@ class Builder implements Serializable {
                         flatten: true,
                         optional: true
                     )
-                    // Archive tap files as a single tar file
-                    context.sh """
-                        cd ${tarDir}/
-                        tar -czf ${tarTap} *.tap
-                    """
-                    try {
-                        context.timeout(time: pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT, unit: 'HOURS') {
-                            context.archiveArtifacts artifacts: "${tarDir}/${tarTap}"
+                    // Archive tap files as a single tar file if we have any
+                    def tapExists = context.sh(script: "ls -l ${tarDir}/*.tap", returnStatus:true)
+                    if (tapExists == 0) {
+                        context.sh """
+                            cd ${tarDir}/
+                            tar -czf ${tarTap} *.tap
+                        """
+                        try {
+                            context.timeout(time: pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT, unit: 'HOURS') {
+                                context.archiveArtifacts artifacts: "${tarDir}/${tarTap}"
```

Comment:
Will this step fail if it wasn't able to create a tarball?

Reply:
this is the tests tap file, so if there is no JDK tarball, there will be no TAP files... hence why I added the check a few lines up to check if there were any... as previously it bombed out the pipeline with an exit code

Reply:
so no is the answer, because we check if one exists.